src/sql/base/browser/ui/table/formatters.ts

```diff
@@ -102,6 +102,15 @@ export function textFormatter(row: number | undefined, cell: any | undefined, va
 	return `<span title="${titleValue}" class="${cellClasses}">${valueToDisplay}</span>`;
 }
 
+
+/**
+ * Format all text to replace all new lines with spaces and performs HTML entity encoding
+ */
+export function iconCssFormatter(row: number | undefined, cell: any | undefined, value: any, columnDef: any | undefined, dataContext: any | undefined): string {
+	let cellClasses = `grid-cell-value-container icon codicon slick-icon-cell-content ${value.text}`;
+	return `<span class="${cellClasses}">&nbsp;</span>`;
```

Comment:
am i missing something here? I don't see how is the aria-label being used here, I don't know whether screen reader will read the span, I think you should use a div and set its role to image and also set the aria-label.

Reply:
it doesn't need to be keyboard focusable, using the screen reader navigation mode will be able to get to it.

Reply:
> am i missing something here? I don't see how is the aria-label being used here, I don't know whether screen reader will read the span, I think you should use a div and set its role to image and also set the aria-label.

I used slickGridDataItemColumnValueWithNoData function for that. It's responsible for creation gridCell. Below is how table row gets rendered:
```
<div role="row" aria-rowindex="1" class="ui-widget-content slick-row even" style="top:0px">
      <div tabindex="-1" role="gridcell" aria-colindex="1" aria-label="Database Icon" class="slick-cell l0 r0">
            <span class="grid-cell-value-container icon codicon slick-icon-cell-content model-view-component-icon-9">&nbsp;</span>
      </div>
      <div tabindex="-1" role="gridcell" aria-colindex="2" aria-label="SNK-PC:AsmtTests" class="slick-cell l1 r1">
            <span title="SNK-PC:AsmtTests" class="grid-cell-value-container">SNK-PC:AsmtTests</span>
      </div>
</div>
```
I decided to use this approach to stay consistent with existing aria-label having elements.

Reply:
the content should have a role 'image', otherwise the screen reader wouldn't know this is an image column

Reply:
please try it out with screen reader and make sure it reads both the aria label and role